openwhisk/executor_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package openwhisk
+
+import (
+	"fmt"
+	"time"
+)
+
+func ExampleNewExecutor_failed() {
+	proc := NewExecutor("true")
+	err := proc.Start()
+	fmt.Println(err)
+	proc.Stop()
+	proc = NewExecutor("/bin/pwd")
+	err = proc.Start()
+	fmt.Println(err)
+	proc.Stop()
+	proc = NewExecutor("donotexist")
+	err = proc.Start()
+	fmt.Println(err)
+	proc.Stop()
+	proc = NewExecutor("/etc/passwd")
+	err = proc.Start()
+	fmt.Println(err)
+	proc.Stop()
+	// Output:
+	// command exited
+	// command exited
+	// command exited
+	// command exited
+}
+
+func ExampleNewExecutor_bc() {
+	proc := NewExecutor("_test/bc.sh")
+	err := proc.Start()
+	fmt.Println(err)
```

Comment:
Should check that err == nil?

Reply:
It is checked. I use the test style "Example" allowed by GO where I check the output.